### ISSUE TITLE
VS Code extension reports errors for valid usage of BindingExtension with x:DataType in MAUI XAML

### ISSUE THREAD

**Issue by simonrozsival** [microsoft/vscode-dotnettools#1131]
Type: <b>Bug</b>

For the following XAML code, VS Code extension will report an error message `XLS0413`. The XAML is correct and this error message should not be displayed. The MAUI app builds correctly, XamlC doesn’t report any errors:
```xml
<Button Text="{Binding Text, x:DataType=Label}" />
```
The error message:
```
The property 'DataType' was not found in type 'BindingExtension'.(XLS0413)
```
Similarly, the extension reports an unsubstantiated warning for the following (unusual) XAML:
```xml
<Button>
  <Button.Text>
    <BindingExtension Path="Text" x:DataType="Label" />
  </Button.Text>
</Button>
```
The warning message:
```
The attribute 'DataType' from the XAML namespace is not defined. (XLS1102)
```
Contrary to expectations, the following XAML, which isn’t correctly understood by the MAUI XamlC compiler, doesn’t produce any warnings:
```xml
<Button>
  <Button.Text x:DataType="Label">
    <BindingExtension Path="Text" />
  </Button.Text>
</Button>
```

Extension version: 0.11.94
VS Code version: Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:14:32.757Z)
OS version: Darwin arm64 23.4.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Max (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|4, 7, 7|
|Memory (System)|32.00GB (0.30GB free)|
|Process Argv|/Users/simonrozsival/Projects/dogfooding/MyDailyPriorities --crash-reporter-id d2e2bec8-5081-4be8-b73a-9e3b61ff94de|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
tftest:31042121
vstes627:30244334
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
chatpanelc:31048052
dsvsc021:30996838
724cj586:31013169
pythoncenvpt:31049070
fchga941:31046352
pythonprc:31047982
dwnewjupytercf:31046870
26j00206:31048877

```

</details>

**Comment by mgoertz-msft** [comment]
```xaml
<Button Text="{Binding Text, x:DataType=Label}" />
```
Verified that this is fixed in the latest C# extension `v2.34.12`

**Comment by simonrozsival** [comment]
I confirm the issue is resolved. Thank you very much!
